pallet-logion-loc/src/lib.rs

```diff
@@ -155,6 +221,125 @@ pub struct LegalOfficerCase<AccountId, Hash, LocId, BlockNumber, EthereumAddress
     legal_fee: Option<Balance>,
 }
 
+impl<AccountId, Hash, LocId, BlockNumber, EthereumAddress, SponsorshipId, Balance>
+LegalOfficerCase<AccountId, Hash, LocId, BlockNumber, EthereumAddress, SponsorshipId, Balance>
+where
+    AccountId: PartialEq + Clone,
+    Hash: PartialEq + Copy + Ord,
+    LocId: PartialEq + Copy + Ord,
+    EthereumAddress: PartialEq + Clone,
+{
+
+    pub fn ensure_can_add<T: pallet::Config>(&self, items: &ItemsParams<LocId, AccountId, EthereumAddress, Hash>) -> Result<(), sp_runtime::DispatchError> {
+        self.ensure_requester_submits::<T>(&items)?;
+        self.ensure_can_add_metadata::<T>(&items.metadata)?;
+        self.ensure_can_add_files::<T>(&items.files)?;
+        self.ensure_can_add_links::<T>(&items.links)?;
+        Ok(())
+    }
+
+    pub fn ensure_requester_submits<T: pallet::Config>(&self, items: &ItemsParams<LocId, AccountId, EthereumAddress, Hash>) -> Result<(), sp_runtime::DispatchError> {
+        if items.metadata.iter().find(|item| !self.is_requester(&item.submitter)).is_some()
+            || items.files.iter().find(|item| !self.is_requester(&item.submitter)).is_some()
+            || items.links.iter().find(|item| !self.is_requester(&item.submitter)).is_some() {
+            Err(Error::<T>::CannotSubmit)?
+        }
+        Ok(())
+    }
```

Comment:
At first glance I thought there was a bug - A verified issuer can be submitter, in this case publication is done by requester.
But I guess you're right, in the specific context of direct LOC opening, the only possible submitter is the requester.

Reply:
As discussed, all items must be submitted by the requester as it directly publishes all data.